checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -4,88 +4,366 @@
+        return generatePreconditionsBasedOnGuards(atm.getAnnotations(), translateItselfToThis);
+    }
+
+    // Given a set of AnnotationMirrors, returns the list of lock expression preconditions
+    // specified in all the @GuardedBy annotations in the set.
+    // Returns an empty set if no such expressions are found.
+    private Set<Pair<String, String>> generatePreconditionsBasedOnGuards(Set<AnnotationMirror> amList, boolean translateItselfToThis) {
+        Set<Pair<String, String>> preconditions = new HashSet<>();
+
+        if (amList != null) {
+            for (AnnotationMirror annotationMirror : amList) {
+
+                if (AnnotationUtils.areSameByClass( annotationMirror, checkerGuardedByClass)) {
+                    if (AnnotationUtils.hasElementValue(annotationMirror, "value")) {
+                        List<String> guardedByValue = AnnotationUtils.getElementValueArray(annotationMirror, "value", String.class, false);
+
+                        for (String lockExpression : guardedByValue) {
+                            if (translateItselfToThis && lockExpression.equals("itself")) {
+                                lockExpression = "this";
+                            }
+                            preconditions.add(Pair.of(lockExpression, LockHeld.class.toString().substring(10 /* "interface " */)));
+                        }
+                    }
+                }
+            }
+        }
+
+        return preconditions;
     }
 
     @Override
     protected void commonAssignmentCheck(AnnotatedTypeMirror varType,
             AnnotatedTypeMirror valueType, Tree valueTree, /*@CompilerMessageKey*/ String errorKey,
-            boolean isLocalVariableAssignement) {
+            boolean isLocalVariableAssignment) {
+
+        Kind valueTreeKind = valueTree.getKind();
+
+        switch(valueTreeKind) {
+            case NEW_CLASS:
+            case NEW_ARRAY:
+                // Avoid issuing warnings for: @GuardedBy(<something>) Object o = new Object();
+                // Do NOT do this if the LHS is @GuardedByBottom.
+                if (!varType.hasAnnotation(GuardedByBottom.class))
+                    return;
+                break;
+            case INT_LITERAL:
+            case LONG_LITERAL:
+            case FLOAT_LITERAL:
+            case DOUBLE_LITERAL:
+            case BOOLEAN_LITERAL:
+            case CHAR_LITERAL:
+            case STRING_LITERAL:
+                // Avoid issuing warnings for: @GuardedBy(<something>) Object o; o = <some literal>;
+                // Do NOT do this if the LHS is @GuardedByBottom.
+                if (!varType.hasAnnotation(GuardedByBottom.class))
+                    return;
+                break;
+            default:
+        }
+
+        boolean skipSubtypeCheck = false;
+
+        // Assigning a value with a @GuardedBy annotation to a variable with a @GuardSatisfied annotation is always
```

Comment:
"always legal" isn't correct.  It's only legal if the locks in the `@GuardedBy` are currently held.